client_restaurant/shared/src/commonMain/kotlin/presentation/base/ErrorType.kt

```diff
@@ -15,6 +15,8 @@ class PermissionDenied : AuthorizationException()
 open class RequestException: BpException()
 class ClientSideException: RequestException()
 class ServerSideException: RequestException()
-class InvalidCredentialsException(val errorMessage: String): RequestException()
+open class InvalidCredentialsException: RequestException()
 class UserNotFoundException(val errorMessage: String): RequestException()
+class InvalidUserNameException(val errorMessage: String): InvalidCredentialsException()
+class InvalidPasswordException(val errorMessage: String): InvalidCredentialsException()
```

Comment:
Why they are inherenting the InvalidCredentialsException?
Why not BpException?

Reply:
As this are related to it!!
Do you think that the best to inhiret from the BpExeption?